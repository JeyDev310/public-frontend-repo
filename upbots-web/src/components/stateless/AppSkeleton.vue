<template>
  <SkeletonWrapper v-bind="$attrs">
    <component :is="skeletonComponent" v-if="skeletonComponent" />

    <template v-else>
      <rect x="0" y="0" rx="15" ry="15" width="100%" height="100%" />
    </template>
  </SkeletonWrapper>
</template>

<script lang="ts">
import { defineComponent, PropType } from 'vue'

import SkeletonWrapper from './skeletons/SkeletonWrapper.vue'
import SkeletonCard from './skeletons/SkeletonCard.vue'
import SkeletonLineChart from './skeletons/SkeletonLineChart.vue'
import SkeletonPieChart from './skeletons/SkeletonPieChart.vue'
import SkeletonTable from './skeletons/SkeletonTable.vue'

type TSkeletonTypes = 'pieChart' | 'lineChart' | 'card' | 'table'

const loaders = Object.freeze({
  card: 'SkeletonCard',
  table: 'SkeletonTable',
  pieChart: 'SkeletonPieChart',
  lineChart: 'SkeletonLineChart'
})

export default defineComponent({
  name: 'AppSkeleton',

  components: {
    SkeletonWrapper,
    SkeletonCard,
    SkeletonLineChart,
    SkeletonPieChart,
    SkeletonTable
  },

  props: {
    type: {
      type: String as PropType<TSkeletonTypes>,
      default: ''
    }
  },

  setup (props) {
    const skeletonComponent = loaders[props.type]

    return { skeletonComponent }
  }
})
</script>
