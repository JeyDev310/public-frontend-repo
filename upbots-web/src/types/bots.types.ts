enum EPerformanceFeeSteps {
  READY = 'READY',
  OPENED = 'OPENED',
  PERFORMED = 'PERFORMED',
}

interface IPerformanceFee {
  paidAmount: number;
  performedStep: EPerformanceFeeSteps;
  remainedAmount: number;
}

interface IRealisedGain {
  btc: number;
  eur: number;
  ubxt: number;
  usd: number;
}

interface IPeriod {
  year: number;
  month: string;
  week: number;
  day: string;
}

enum EMeasuredObjects {
  BOT = 'BOT',
  SUBSCRIPTION = 'SUBSCRIPTION',
}

export interface IPerformanceBotCycles {
  openAt: string;
  botId?: string;
  botVer?: string;
  subBotId?: string;
  userId?: string;
  measuredObject: EMeasuredObjects;
  openPeriod: IPeriod;
  closeAt?: string;
  closePeriod: IPeriod;
  stratType: string;
  result: string;
  cycleSequence: number;
  open: boolean;
  sbl: string;
  exch: string;
  realisedGain?: IRealisedGain;
  performanceFee?: IPerformanceFee;
  profitPercentage?: number;
  profitPercentageUC?: number;
  entryPrice: number;
  closePrice?: number;
  qExec?: number;
}

type TBotStratType = 'LONG' | 'SHORT' | 'LONG_SHORT';

type TPositionType = 'fixed' | 'percent'

type TBotAccountType = 'spot' | 'future';

interface IBotAccountAllocated {
  maxamount: number;
  currency: string;
}

export interface ISubscriptedBots {
  id: string;
  botId: string;
  apiKeyRef: string;
  enabled: boolean;
  isOwner: boolean;
  botRunning: boolean;
  accountType: TBotAccountType;
  stratType: TBotStratType;
  createdAt?: string;
  updatedAt?: string;
  cycleSequence: number;
  status: number;
  accountPercent: number;
  positionType: TPositionType;
  positionAmount: number;
  accountLeverage: number;
  contractSize: number;
  baseAmount: number;
  openedQuantity: number;
  quote: string;
  accountAllocated: IBotAccountAllocated;
  error?: string;
  errorAt?: string;
  errorReason?: string;
  picture: {
    mimetype: string;
    data: string;
  };
}

export interface IAlgoBotSubscription {
  id: string;
  botId: string;
  apiKeyRef: string;
  enabled: boolean;
  isOwner: boolean;
  botRunning: boolean;
  stratType: string;
  quantity: number;
  createdAt: string;
  updatedAt: string;
  positionType?: string;
  positionAmount?: number;
}

export interface IPerfSubscriptionCycles {
  closeAt: string;
  closePeriod: IPeriod;
  closePrice: number;
  cycleSequence: number;
  entryPrice: number;
  exch: string;
  measuredObject: EMeasuredObjects;
  open: false;
  openAt: string;
  openPeriod: IPeriod;
  performanceFee: IPerformanceFee;
  profitPercentage: number;
  profitPercentageUC: number;
  qExec: number;
  realisedGain: IRealisedGain;
  result: string;
  sbl: string;
  stratType: string;
  subBotId: string;
  userId: string;
}

interface IChartData {
  labels: string[];
  data: number[];
}

interface ICharts {
  [key: string]: IChartData;
}

type TBotMeasuredObjects = 'BOT' | 'SUBSCRIPTION'

type TBotSizes = 'MONTH6' | 'MONTH12' | 'ALLMONTH'

export interface IBotPerformanceSubscriptionSnapshotById {
  allmonths: number;
  allmonthsUC: number;
  botId: string;
  charts: ICharts;
  fees: number;
  maxDrawdown: number;
  measuredObject: TBotMeasuredObjects;
  month1: number;
  month1UC: number;
  month3: number;
  month3UC: number;
  month6: number;
  month6UC: number;
  size: TBotSizes;
  snapshotDate: string;
  stratType: TBotStratType;
  subBotId: string;
}

export enum EWalletStatuses {
  ENABLED = 'ENABLED',
  DISABLED = 'DISABLED',
  LOCKED = 'LOCKED',
}

export interface IPerfeesBotWallets {
  userId: string;
  botId: string;
  botSubId: string;
  amount: number;
  allocatedAmount: number;
  creditAmount: number;
  debtAmount: number;
  paidAmount: number;
  status: EWalletStatuses;
  autoRefill: boolean;
  botRef?: string;
  botName?: string;
  name?: string;
}

interface IEventEntity {
  at: string;
  status: string;
  exOrderId: string;
  side: string;
  sbl: string;
  exch: string;
  type: string;
  qOrig: string;
  qExec: string;
  qRem: string;
  qExecCumul: string;
  source: string;
  pAsk: string;
  pExec: string;
  cumulQuoteCost: string;
}

interface ICompletionEntity {
  qExec: number;
  pExec: number;
  cumulQuoteCost: number;
}

export enum EUserFriendlyOrderStatus {
  NA = 'na',
  CANCELED = 'cancelled',
  FILLED = 'filled',
  PARTIAL = 'partial',
  WAITING = 'waiting',
  FAILED = 'failed',
}

interface IOrderTrackingModel {
  id: string;
  botId: string;
  userId: string;
  exchKeyId: string;
  botSubId: string;
  stratId: string;
  ctxBot: string;
  ctx: string;
  side: string;
  orderType: string;
  priceAsked: string;
  qtyBaseAsked: string;
  qtyQuoteAsked?: any;
  sbl: string;
  exch: string;
  initiator: string;
  aborted: boolean;
  completed: boolean;
  cycleSequence: number;
  signalId: string;
  completion: ICompletionEntity;
  errorAt: string;
  errorReason: string;
  events: IEventEntity[];
  // eslint-disable-next-line camelcase
  created_at: string;
  // eslint-disable-next-line camelcase
  updated_at: string;
  userOrderStatus: EUserFriendlyOrderStatus;
}

export interface IAlgobotsSubscriptionAudits {
  botId: string;
  botSubId: string;
  userId: string;
  oTrackId: IOrderTrackingModel;
  accountPercent: number;
  status: number;
  position: string;
  followed: boolean;
  cycleSequence: number;
  createdAt: string;
  updatedAt: string;
  errorReason: string;
  errorAt: { type: string; };
  signalId: { type: string; };
  currency: { type: string; };
  exchange: { type: string; };
  balance: number;
}

export interface IPerfeesUserWallet {
  userId: string;
  amount: number;
  allocatedAmount: number;
  availableAmount: number;
  debtAmount: number;
  creditAmount: number;
  rewardCreditedForBots: boolean;
  depositAddressETH: string;
  depositAddressBSC: string;
  referral?: {
    myCode: string;
    invitedCode: string;
    totalEarned: number;
  };
}

interface IBotPerFeesDistr {
  burn: number;
  developer: number;
  pool: number;
  reserve: number;
  percent: number;
}

interface IBotPerfFees {
  address: string;
  distribution: IBotPerFeesDistr;
  percent: number;
}

interface IBotPerfSnapshots {
  allmonths: number;
  botId: string;
  charts: ICharts;
  fees: number;
  maxDrawdown: number;
  measuredObject: TBotMeasuredObjects;
  month1: number;
  month3: number;
  month6: number;
  size: TBotSizes;
  snapshotDate: string;
  stratType: TBotStratType;
}

interface IPicture {
  mimetype: string;
  data: string;
}

interface IRaters {
  id: string;
  user: string;
  vote: number;
  comment: string;
  reviewedAt: string;
}

export interface IWebhook {
  exitTrigger: boolean;
}

export type TCategory = 'algobot' | 'copybot' | 'userbot'

export enum ECategory {
  AlgoBot = 'algobot',
  CopyBot = 'copybot',
  UserBot = 'userbot'
}

export interface IBotSnapshot {
  id: string;
  name: string;
  botVer: string;
  description: string;
  base: string;
  quote: string;
  stratType: TBotStratType;
  category: TCategory;
  ownerId: string;
  updatedAt?: string;
  creator: string;
  picture: IPicture;
  avgtrades: number;
  lastMonthTrades: number;
  allocatedMaxAmount: number;
  allocatedCurrency: string;
  ratings: number;
  raters?: IRaters[];
  reviewerName: string;
  reviewerImg: string;
  reviewerBotRating: number;
  botRef: string;
  perfFees: IBotPerfFees;
  realOwnerId?: string;
  exchangesType: string[];
  priceDecimal: number;
  webhook: IWebhook;
  img: string;
  perfSnapshots: IBotPerfSnapshots;
}

export interface IBotSubscription {
  id: string;
  botId: string;
  apiKeyRef: string;
  enabled: true;
  isOwner: boolean;
  botRunning: boolean;
  accountType: TBotAccountType;
  stratType: TBotStratType;
  createdAt?: string;
  updatedAt?: string;
  cycleSequence: number;
  status: number;
  accountPercent: number;
  accountLeverage: number;
  contractSize: number;
  baseAmount: number;
  openedQuantity: number;
  accountAllocated: IBotAccountAllocated;
  positionType?: string;
  positionAmount?: number;
  quote?: string;
  error?: string;
  errorAt?: string;
  errorReason?: string;
}

// NOT USED FOR NOW
// export interface IPopularBot {
//   botId: string;
//   botName: string[];
//   followers: number;
//   amount: number;
// }

export interface IFollowBotRequest {
  botId: string;
  apiKeyRef: string;
  // quantity: number;
  // FIXME: only percentage should be sent (quantity is old specs)
  accountType: string;
  stratType: string;
  accountPercentage: number;
  accountLeverage?: number;
  contractSize?: number;
  baseAmount?: number;
  positionType?: string;
  positionAmount?: number;
}

export interface IRefill {
  userId: string;
  botId: string;
  botSubId: string;
  amount: number;
  allocatedAmount: number;
  creditAmount: number;
  debtAmount: number;
  paidAmount: number;
  status: EWalletStatuses;
  autoRefill: boolean;
  createdAt: string;
}

export interface IReviewRecord {
  user: string;
  picture: {
    mimetype: string;
    data: string;
  };
  vote: number;
  comment: string;
  reviewedAt: number; // valude of Date.getTime()
}

export enum EPositionType {
  FIXED = 'fixed',
  PERCENT = 'percent'
}

export enum EAlgoBotsCardType {
  Bot = 'bot',
  BestBot = 'best-bot',
  PopularBot = 'popular-bot'
}

export type TStratType = 'LONG' | 'SHORT' | 'LONG_SHORT'

export enum EBotStratType {
  LONG = 'LONG',
  SHORT = 'SHORT',
  LongShort = 'LONG_SHORT',
}

export type TOrderType = 'MARKET' | 'LIMIT'

export enum EBotSnapshotRef {
  SURGEBOTSXP = 'SURGEBOTSXP',
  SUPERTRADERLINK = 'SUPERTRADERLINK',
  SUPERTRADERBNB = 'SUPERTRADERBNB',
  HARSIBTC = 'HARSIBTC',
  AVAXUSDT1 = 'AVAXUSDT1',
  TOMOLO1 = 'TOMOLO1'
}

export enum EStrategy {
  LONG = 'long',
  SHORT = 'short',
  LongShort = 'long-short'
}

export interface IBotUpdate {
  name?: string;
  botVer?: string;
  description?: string;
  base?: string;
  quote?: string;
  stratType?: TBotStratType;
  category?: TCategory;
  ownerId?: string;
  avgtrades?: number;
}

interface IAllocated {
  username: string;
  userimg: string;
  botrating: number;
}

interface IMarket {
  base: string;
  quote: string;
}

interface IPercent {
  percent: number;
}

interface IReviews {
  username: string;
  userimg: string;
  botrating: number;
}

export interface IUpdateBotInfo {
  allocated: IAllocated;
  avgtrades: number;
  botRef: string;
  botVer: string;
  category: TCategory;
  createdAt: string;
  creator: string;
  description: string;
  enabled: boolean;
  exchangesType: string[];
  lastMonthTrades: number;
  market: IMarket;
  name: string;
  owner: string;
  perfFees: IPercent;
  priceDecimal: number;
  raters: string[];
  ratings: number;
  reviews: IReviews;
  stratType: TStratType;
  updatedAt: string;
  webhook: IWebhook;
}

export interface IUserBotWebhook {
  botId: string;
  stratType: string;
  position: string;
  userFilter?: string[];
  secret?: string;
}

export interface IUserBotWebhookMessage {
  webhookURL: string;
  messages: IUserBotWebhook[];
}

export enum EBotSorted {
  Performance = 'performance',
  Performance6M = 'performance_6m',
  Performance3M = 'performance_3m',
  Date = 'date'
}

export enum EBotStrategy {
  All = 'all'
}

export enum EBotStatus {
  Active = 'active',
  Inactive = 'inactive',
  Paused = 'paused',
}
