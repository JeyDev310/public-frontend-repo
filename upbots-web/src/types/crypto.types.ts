export interface ICoinPriceData {
  // eslint-disable-next-line camelcase
  market_caps: [number, number][];
  prices: [number, number][];
  // eslint-disable-next-line camelcase
  total_volumes: [number, number][];
}

export interface ICoinPriceResponsePayload {
  cryptoSymbol: string;
  fiatSymbol: string;
  data: ICoinPriceData;
}

export interface ICoinItem {
  title: string;
  name: string;
  symbol: string;
  srcCoin: string;
}

export interface ICoinPriceRequestPayload {
  cryptoSymbol: string;
  fiatSymbol: string;
}

interface IPriceChange {
  percentage: number;
  absolute: number;
}

interface ICurrentPrice {
  last: number;
  high: number;
  low: number;
  change: IPriceChange;
}

interface ICurrentPriceSummary {
  price: ICurrentPrice;
  volume: number;
  volumeQuote: number;
}

export interface ICurrentPriceSummaryResult {
  result: ICurrentPriceSummary;
}

export enum ECryptoCurrency {
  BTC = 'BTC',
  ETH = 'ETH',
  BUSD = 'BUSD',
  BNB = 'BNB',
  AVAX = 'AVAX',
  UBXT = 'UBXT',
  LINK = 'LINK',
  TOMO = 'TOMO',
  USDT = 'USDT',
  XBT = 'XBT',
  FTT = 'FTT',
  ETC = 'ETC',
  BAT = 'BAT',
  CAKE = 'CAKE',
  SHIB = 'SHIB'
}

export enum EAccount {
  BINANCE = 'binance',
  KUCOIN = 'kucoin',
  HUOBI = 'huobi',
  FTX = 'ftx',
  COINBASEPRO = 'coinbasepro',
  BinanceUS = 'binance-us',
  KucoinFuture = 'kucoin-future',
  BinanceFuture = 'binance-future',
  FtxFuture = 'ftx-future'
}

export enum EAccountType {
  SPOT = 'spot',
  FUTURE = 'future'
}

export interface ICryptoPriceLatest {
  time: number;
  high: number;
  low: number;
  open: number;
  volumefrom: number; // volumeBase
  volumeto: number; // volumeQuote
  close: number;
  conversionType: string;
  conversionSymbol: string;
}

export interface ICryptoPrice {
  Response: string;
  Message: string;
  HasWarning: boolean;
  Type: number;
  Ignored?: boolean;
  RateLimit: any;
  // latest tick on an hourly base
  Latest: ICryptoPriceLatest;
  // latest tick on the previous day based on CEST time
  LatestPreviousDay: ICryptoPriceLatest;
  Data: {
    Aggregated: boolean;
    TimeFrom: number;
    TimeTo: number;
    HourlyData: ICryptoPriceLatest[];
    DailyData: ICryptoPriceLatest[];
  };
}
