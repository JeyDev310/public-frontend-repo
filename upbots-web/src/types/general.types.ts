export interface ITableHeaders {
  property: string;
  label: string;
  sortable?: boolean;
  show?: boolean;
  cellClasses?: string;
  width?: number;
  minWidth?: number;
}

export interface ITableSort {
  order: string;
  orderBy: string;
}

export type TIndexedObject<T = any> = {
  [key: string]: T;
}

export type TNullableField<T> = T | null

export type TIndexedArray<T> = Array<T | any>

export type TCallback = <T>(t?: T) => void;

export type TCallbackFn<T extends unknown[] = [], R = void> = (...args: T) => R

export type TSidebar = {
  route: string;
  label: string;
  icon: string;
  isComingSoon?: boolean;
  betaLabel?: string;
  isBeta?: boolean;
  exact: boolean;
  isMobilePreview?: boolean;
  subListItems?: TSidebar[];
  isDAppPage?: boolean;
}

export type TDropdownOption = {
  label: string;
  value: string | number;
}
