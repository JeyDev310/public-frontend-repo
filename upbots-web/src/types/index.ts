export * from './general.types'
export * from './user.types'
export * from './transaction.types'
export * from './ubxt.types'
export * from './bots.types'
export * from './account/account.types'
export * from './errors/errors.types'

/* COMPONENTS */
export * from '@/components/stateless/charts/chart.types'
export * from '@/components/stateless/notifications/notification.types'

/* VIEW */
export * from '@/views/auth/auth.types'
export * from '@/views/profile/profile.types'
export * from '@/views/notifications/notifications.types'
export * from '@/views/keys/keys.types'
export * from '@/views/portfolio/portfolio.types'
export * from '@/views/portfolio/store/cex/cex.types'
export * from '@/views/portfolio/store/dex/dex.types'
export * from '@/views/swap/swap.types'
export * from '@/views/ubxt-wallet/ubxt-wallet.types'
export * from '@/views/staking/staking.types'
export * from '@/views/my-trades/my-trades.types'
