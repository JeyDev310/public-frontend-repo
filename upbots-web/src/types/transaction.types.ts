export enum ETransactionTypes {
  TRANSFER = 'TRANSFER',
  DEPOSIT = 'DEPOSIT',
  WITHDRAW = 'WITHDRAW',
  PERFORMANCE_FEE = 'PERFORMANCE_FEE',
}

export enum ETransactionStatuses {
  PENDING = 'PENDING',
  TRANSFERRING = 'TRANSFERRING',
  COMPLETED = 'COMPLETED',
  FAILED = 'FAILED',
  CONFIRMED = 'CONFIRMED',
}

enum EFeeRecipientType {
  DEVELOPER = 'DEVELOPER',
  BURN = 'BURN',
  POOL = 'POOL',
  RESERVE = 'RESERVE',
  GROUP = 'GROUP',
}

export interface IUserTransaction {
  transactionId?: string;
  userId?: string;
  botId: string;
  botName?: string;
  type: ETransactionTypes;
  subType?: string;
  completed?: boolean;
  status: ETransactionStatuses;
  address?: string;
  amount: number;
  totalAmount?: number;
  hash?: string;
  group?: EFeeRecipientType;
  confirmations?: number;
  confirmPercent?: number;
  explorer?: string;
  error?: string;
  extra?: any;
}
