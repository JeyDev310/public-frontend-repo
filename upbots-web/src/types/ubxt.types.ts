export enum EUBXTId {
  UBXTID = 'upbots',
}
