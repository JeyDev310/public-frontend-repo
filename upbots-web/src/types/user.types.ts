import { TDropdownOption } from '@/types'

export enum ECouponTypes {
  UNIQUE = 'unique',
  GLOBAL = 'global',
}
export interface ICoupons {
  code: string;
  promoName: string;
  description: string;
  couponType: ECouponTypes;
  activated: boolean;
  activationDate?: Date;
  validFrom?: Date;
  validTo?: Date;
  userId?: string;
  createdAt?: Date;
  updatedAt?: Date;
}

export interface IExchangeKey {
  id: string;
  name: string;
  created: string;
  exchange: string;
  type: string;
  quoteCurrency?: string;
  publicKey: string;
  valid: boolean;
  lastHealthCheck: string;
  updated: string;
  displayName?: string;
  img?: string;
}
export interface IFavoriteCurrency {
  value: string;
  label: string;
}

export interface IAlgobotFilterSetting {
  status: TDropdownOption;
  strategy: TDropdownOption;
  exchanges: string[];
  pairs: string[];
  sortedByValue: TDropdownOption;
}

export interface IUserSettings {
  userId?: string;
  darkMode?: boolean;
  favoriteCurrency?: IFavoriteCurrency;
  algobotFilters?: IAlgobotFilterSetting;
}
