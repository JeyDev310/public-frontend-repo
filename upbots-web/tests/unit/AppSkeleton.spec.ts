import { mount, MountingOptions } from '@vue/test-utils'
import { TIndexedObject } from '@/types'
import AppSkeleton from '@/components/stateless/AppSkeleton.vue'

const factory = (mountOptions?: MountingOptions<TIndexedObject>) => mount(AppSkeleton, mountOptions)

describe('AppSkeleton.vue', () => {
  it('"default" prop definition', () => {
    expect(AppSkeleton.props).toHaveProperty('type.default')
    expect(AppSkeleton.props.type.default).toBe('')
    expect(AppSkeleton.props.type.type).toBe(String)
  })

  it('mount without props', () => {
    const wrapper = factory()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('mount type "pieChart"', () => {
    const wrapper = factory({
      props: { type: 'pieChart' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('mount type "lineChart"', () => {
    const wrapper = factory({
      props: { type: 'lineChart' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('mount type "card"', () => {
    const wrapper = factory({
      props: { type: 'card' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('mount type "table"', () => {
    const wrapper = factory({
      props: { type: 'table' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
