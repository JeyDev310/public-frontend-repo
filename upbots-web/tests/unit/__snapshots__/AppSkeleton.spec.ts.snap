// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSkeleton.vue mount type "card" 1`] = `
<div><svg width="100%" height="100%" viewBox="0 0 400 200" preserveAspectRatio="xMidYMid meet" version="1.1">
    <rect style="fill: url(#6kgjdkwyvd4);" clip-path="url(#husfxph3u2o)" x="0" y="0" width="100%" height="100%"></rect>
    <defs>
      <clipPath id="husfxph3u2o">
        <rect x="50%" y="20" rx="8" width="190" height="14" transform="translate(-95)"></rect>
        <rect x="50%" y="45" rx="8" width="120" height="12" transform="translate(-60)"></rect>
        <rect x="60" y="80" rx="8" width="70" height="12"></rect>
        <rect x="60" y="100" rx="8" width="100" height="12"></rect>
        <rect x="100%" y="80" rx="8" width="100" height="12" transform="translate(-160)"></rect>
        <rect x="100%" y="100" rx="8" width="60" height="12" transform="translate(-160)"></rect>
        <rect x="60" y="130" rx="8" width="90" height="12"></rect>
        <rect x="60" y="150" rx="8" width="100" height="12"></rect>
        <rect x="100%" y="130" rx="8" width="100" height="12" transform="translate(-160)"></rect>
        <rect x="100%" y="150" rx="8" width="80" height="12" transform="translate(-160)"></rect>
      </clipPath>
      <linearGradient id="6kgjdkwyvd4">
        <stop offset="0%" stop-color="#E4EEF3" stop-opacity="1">
          <animate attributeName="offset" values="-2; 1" dur="1s" repeatCount="indefinite"></animate>
        </stop>
        <stop offset="50%" stop-color="#a7c7d8" stop-opacity="1">
          <animate attributeName="offset" values="-1.5; 1.5" dur="1s" repeatCount="indefinite"></animate>
        </stop>
        <stop offset="100%" stop-color="#E4EEF3" stop-opacity="1">
          <animate attributeName="offset" values="-1; 2" dur="1s" repeatCount="indefinite"></animate>
        </stop>
      </linearGradient>
    </defs>
  </svg></div>
`;

exports[`AppSkeleton.vue mount type "lineChart" 1`] = `
<div><svg width="100%" height="100%" viewBox="0 0 400 200" preserveAspectRatio="xMidYMid meet" version="1.1">
    <rect style="fill: url(#vipg4xf7rcn);" clip-path="url(#d0kw5tmsi8c)" x="0" y="0" width="100%" height="100%"></rect>
    <defs>
      <clipPath id="d0kw5tmsi8c">
        <rect x="8" y="10" rx="0" ry="0" width="2" height="180"></rect>
        <rect x="8" y="190" rx="0" ry="0" width="380" height="2"></rect>
        <path d="M20,60 C240,240 190,-30 380.00,50 L380,180 L20,180 Z"></path>
      </clipPath>
      <linearGradient id="vipg4xf7rcn">
        <stop offset="0%" stop-color="#E4EEF3" stop-opacity="1">
          <animate attributeName="offset" values="-2; 1" dur="1s" repeatCount="indefinite"></animate>
        </stop>
        <stop offset="50%" stop-color="#a7c7d8" stop-opacity="1">
          <animate attributeName="offset" values="-1.5; 1.5" dur="1s" repeatCount="indefinite"></animate>
        </stop>
        <stop offset="100%" stop-color="#E4EEF3" stop-opacity="1">
          <animate attributeName="offset" values="-1; 2" dur="1s" repeatCount="indefinite"></animate>
        </stop>
      </linearGradient>
    </defs>
  </svg></div>
`;

exports[`AppSkeleton.vue mount type "pieChart" 1`] = `
<div><svg width="100%" height="100%" viewBox="0 0 400 200" preserveAspectRatio="xMidYMid meet" version="1.1">
    <rect style="fill: url(#7wit26ja9it);" clip-path="url(#hf3cml38l5d)" x="0" y="0" width="100%" height="100%"></rect>
    <defs>
      <clipPath id="hf3cml38l5d">
        <circle cx="240" cy="100" r="80"></circle>
        <rect x="60" y="40" rx="10" ry="10" width="10" height="10"></rect>
        <rect x="80" y="40" rx="5" ry="10" width="40" height="10"></rect>
        <rect x="60" y="55" rx="10" ry="10" width="10" height="10"></rect>
        <rect x="80" y="55" rx="5" ry="10" width="40" height="10"></rect>
        <rect x="60" y="70" rx="10" ry="10" width="10" height="10"></rect>
        <rect x="80" y="70" rx="5" ry="10" width="40" height="10"></rect>
        <rect x="60" y="85" rx="10" ry="10" width="10" height="10"></rect>
        <rect x="80" y="85" rx="5" ry="10" width="40" height="10"></rect>
      </clipPath>
      <linearGradient id="7wit26ja9it">
        <stop offset="0%" stop-color="#E4EEF3" stop-opacity="1">
          <animate attributeName="offset" values="-2; 1" dur="1s" repeatCount="indefinite"></animate>
        </stop>
        <stop offset="50%" stop-color="#a7c7d8" stop-opacity="1">
          <animate attributeName="offset" values="-1.5; 1.5" dur="1s" repeatCount="indefinite"></animate>
        </stop>
        <stop offset="100%" stop-color="#E4EEF3" stop-opacity="1">
          <animate attributeName="offset" values="-1; 2" dur="1s" repeatCount="indefinite"></animate>
        </stop>
      </linearGradient>
    </defs>
  </svg></div>
`;

exports[`AppSkeleton.vue mount type "table" 1`] = `
<div><svg width="100%" height="100%" viewBox="0 0 400 200" preserveAspectRatio="xMidYMid meet" version="1.1">
    <rect style="fill: url(#o5uc01tei7r);" clip-path="url(#5h9jn2itlm8)" x="0" y="0" width="100%" height="100%"></rect>
    <defs>
      <clipPath id="5h9jn2itlm8">
        <rect x="51" y="45" rx="3" ry="3" width="906" height="17"></rect>
        <circle cx="879" cy="123" r="11"></circle>
        <circle cx="914" cy="123" r="11"></circle>
        <rect x="104" y="115" rx="3" ry="3" width="141" height="15"></rect>
        <rect x="305" y="114" rx="3" ry="3" width="299" height="15"></rect>
        <rect x="661" y="114" rx="3" ry="3" width="141" height="15"></rect>
        <rect x="55" y="155" rx="3" ry="3" width="897" height="2"></rect>
        <circle cx="880" cy="184" r="11"></circle>
        <circle cx="915" cy="184" r="11"></circle>
        <rect x="105" y="176" rx="3" ry="3" width="141" height="15"></rect>
        <rect x="306" y="175" rx="3" ry="3" width="299" height="15"></rect>
        <rect x="662" y="175" rx="3" ry="3" width="141" height="15"></rect>
        <rect x="56" y="216" rx="3" ry="3" width="897" height="2"></rect>
        <circle cx="881" cy="242" r="11"></circle>
        <circle cx="916" cy="242" r="11"></circle>
        <rect x="106" y="234" rx="3" ry="3" width="141" height="15"></rect>
        <rect x="307" y="233" rx="3" ry="3" width="299" height="15"></rect>
        <rect x="663" y="233" rx="3" ry="3" width="141" height="15"></rect>
        <rect x="57" y="274" rx="3" ry="3" width="897" height="2"></rect>
        <circle cx="882" cy="303" r="11"></circle>
        <circle cx="917" cy="303" r="11"></circle>
        <rect x="107" y="295" rx="3" ry="3" width="141" height="15"></rect>
        <rect x="308" y="294" rx="3" ry="3" width="299" height="15"></rect>
        <rect x="664" y="294" rx="3" ry="3" width="141" height="15"></rect>
        <rect x="58" y="335" rx="3" ry="3" width="897" height="2"></rect>
        <circle cx="881" cy="363" r="11"></circle>
        <circle cx="916" cy="363" r="11"></circle>
        <rect x="106" y="355" rx="3" ry="3" width="141" height="15"></rect>
        <rect x="307" y="354" rx="3" ry="3" width="299" height="15"></rect>
        <rect x="663" y="354" rx="3" ry="3" width="141" height="15"></rect>
        <rect x="57" y="395" rx="3" ry="3" width="897" height="2"></rect>
        <circle cx="882" cy="424" r="11"></circle>
        <circle cx="917" cy="424" r="11"></circle>
        <rect x="107" y="416" rx="3" ry="3" width="141" height="15"></rect>
        <rect x="308" y="415" rx="3" ry="3" width="299" height="15"></rect>
        <rect x="664" y="415" rx="3" ry="3" width="141" height="15"></rect>
        <rect x="55" y="453" rx="3" ry="3" width="897" height="2"></rect>
        <rect x="51" y="49" rx="3" ry="3" width="2" height="465"></rect>
        <rect x="955" y="49" rx="3" ry="3" width="2" height="465"></rect>
        <circle cx="882" cy="484" r="11"></circle>
        <circle cx="917" cy="484" r="11"></circle>
        <rect x="107" y="476" rx="3" ry="3" width="141" height="15"></rect>
        <rect x="308" y="475" rx="3" ry="3" width="299" height="15"></rect>
        <rect x="664" y="475" rx="3" ry="3" width="141" height="15"></rect>
        <rect x="52" y="513" rx="3" ry="3" width="904" height="2"></rect>
        <rect x="52" y="80" rx="0" ry="0" width="904" height="17"></rect>
        <rect x="52" y="57" rx="3" ry="3" width="68" height="33"></rect>
        <rect x="222" y="54" rx="3" ry="3" width="149" height="33"></rect>
        <rect x="544" y="55" rx="3" ry="3" width="137" height="33"></rect>
        <rect x="782" y="56" rx="3" ry="3" width="72" height="33"></rect>
        <rect x="933" y="54" rx="3" ry="3" width="24" height="33"></rect>
      </clipPath>
      <linearGradient id="o5uc01tei7r">
        <stop offset="0%" stop-color="#E4EEF3" stop-opacity="1">
          <animate attributeName="offset" values="-2; 1" dur="1s" repeatCount="indefinite"></animate>
        </stop>
        <stop offset="50%" stop-color="#a7c7d8" stop-opacity="1">
          <animate attributeName="offset" values="-1.5; 1.5" dur="1s" repeatCount="indefinite"></animate>
        </stop>
        <stop offset="100%" stop-color="#E4EEF3" stop-opacity="1">
          <animate attributeName="offset" values="-1; 2" dur="1s" repeatCount="indefinite"></animate>
        </stop>
      </linearGradient>
    </defs>
  </svg></div>
`;

exports[`AppSkeleton.vue mount without props 1`] = `
<div><svg width="100%" height="100%" viewBox="0 0 400 200" preserveAspectRatio="xMidYMid meet" version="1.1">
    <rect style="fill: url(#laqo9kgi19l);" clip-path="url(#40ce5w19dyx)" x="0" y="0" width="100%" height="100%"></rect>
    <defs>
      <clipPath id="40ce5w19dyx">
        <rect x="0" y="0" rx="5" ry="5" width="100%" height="100%"></rect>
      </clipPath>
      <linearGradient id="laqo9kgi19l">
        <stop offset="0%" stop-color="#E4EEF3" stop-opacity="1">
          <animate attributeName="offset" values="-2; 1" dur="1s" repeatCount="indefinite"></animate>
        </stop>
        <stop offset="50%" stop-color="#a7c7d8" stop-opacity="1">
          <animate attributeName="offset" values="-1.5; 1.5" dur="1s" repeatCount="indefinite"></animate>
        </stop>
        <stop offset="100%" stop-color="#E4EEF3" stop-opacity="1">
          <animate attributeName="offset" values="-1; 2" dur="1s" repeatCount="indefinite"></animate>
        </stop>
      </linearGradient>
    </defs>
  </svg></div>
`;
